cmd/scollector/collectors/vsphere.go

```diff
@@ -61,7 +61,12 @@ func c_vsphere(user, pwd, vHost string, cpuIntegrators map[string]tsIntegrator)
 
 	// Data for Hosts (Hypervisors)
 	for _, host := range hostSystems {
-		name := util.Clean(host.Name)
+		name, err := util.GetHostManager().GetNameProcessor().FormatName(host.Name)
```

Comment:
Yes this was intended and the reason I oddly chose to tick both _breaking_ and _non-breaking_ change in PR message.

Prior to now the Bosun hostname always comes from a call to `os.Hostname()` and this should always be valid.

It has been possible to override the hostname with SCollector however I can't imagine why anyone would have wanted to specify an invalid (according to IETF rules) hostname and if an invalid name were provided then it could end up causing problems elsewhere which would be much harder to debug than if a clear error was raised if an invalid hostname was provided.